app/policies/address_policy.rb

```diff
@@ -0,0 +1,32 @@
+class AddressPolicy < ApplicationPolicy
+  include PoliciesHelper
+
+  def new?
+    return true if user.admin?
+
+    case record.addressable
```

Comment:
This is a smell to me.  You shouldn't have to create the `PoliciesHelper` class to handle this.

1. is there any case when this record would _not_ be addressable?  (Since the Address class has `validates_presence_of :addressable` then it must always be true, yes?)
2. can you just ask "addressable" if this user can edit this address?
    So a Company will know how to respond to that by checking if the user is a member.  Other (future) classes will handle this using whatever business logic is appropriate.

Reply:
1. No, there is no current use case where a record would not be addressable.  I am using case statement here in anticipation of adding a `when User` clause when we extend Address to be members.

2. Yes, for the current situation what you suggest would work fine.  However, with member address this kind of structure will be necessary.

Reply:
Can we instead replace this with:
```
def new?
   user.admin? || is_in_company?(record)
end
```
which is the same code that is in CompanyPolicy .update?